src/test/regress/sql/create_drop_database_propagation.sql

```diff
@@ -0,0 +1,121 @@
+
+
+\set create_drop_db_tablespace :abs_srcdir '/tmp_check/ts3'
+CREATE TABLESPACE create_drop_db_tablespace LOCATION :'create_drop_db_tablespace';
+
+\c - - - :worker_1_port
+\set create_drop_db_tablespace :abs_srcdir '/tmp_check/ts4'
+CREATE TABLESPACE create_drop_db_tablespace LOCATION :'create_drop_db_tablespace';
+
+\c - - - :worker_2_port
+\set create_drop_db_tablespace :abs_srcdir '/tmp_check/ts5'
+CREATE TABLESPACE create_drop_db_tablespace LOCATION :'create_drop_db_tablespace';
+
+\c - - - :master_port
+create user create_drop_db_test_user;
+
+set citus.enable_create_database_propagation=on;
+
+CREATE DATABASE mydatabase
```

Comment:
For some reason CREATE DATABASE is not propagated when adding a new node.
If you run the following in a `citus_dev` created cluster:
```
select citus_remove_node('localhost', 9702);
set citus.enable_create_database_propagation = true;
create database abc;
set citus.log_remote_commands = true;
set citus.grep_remote_commands = '%abc%';
select citus_add_node('localhost', 9702);
```
You get this error where it tries to alter the database, without first creating it.
```
psql:create_database_add_node.sql:6: NOTICE:  00000: issuing ALTER DATABASE abc OWNER TO jelte;
DETAIL:  on server jelte@localhost:9702 connectionId: 2
LOCATION:  LogRemoteCommand, remote_commands.c:353
psql:create_database_add_node.sql:6: WARNING:  3D000: database "abc" does not exist
LOCATION:  ReportResultError, remote_commands.c:317
psql:create_database_add_node.sql:6: ERROR:  XX000: failure on connection marked as essential: localhost:9702
LOCATION:  MarkRemoteTransactionFailed, remote_transaction.c:840
```

Reply:
need to create the database objects via SendNodeWideObjectsSyncCommands too

Reply:
Same problem happens in activate_node too:

```sql


-- not needed in regression tests
CREATE OR REPLACE FUNCTION wait_until_metadata_sync(timeout INTEGER DEFAULT 15000)
    RETURNS void
    LANGUAGE C STRICT
    AS 'citus';

-- not needed in regression tests
CREATE OR REPLACE FUNCTION pg_catalog.citus_disable_node_and_wait(nodename text, nodeport integer, force bool DEFAULT false)
    RETURNS void
  LANGUAGE plpgsql
AS $function$
BEGIN

  PERFORM pg_catalog.citus_disable_node(nodename, nodeport, force);
  PERFORM public.wait_until_metadata_sync(30000);
END;
$function$;

select citus_disable_node_and_wait('localhost', 9701, true);

set citus.enable_create_database_propagation=on;
create database onur;

select citus_activate_node('localhost', 9701);
WARNING:  database "onur" does not exist
ERROR:  failure on connection marked as essential: localhost:9701
```